src/common/libpkcs11.c

```diff
@@ -99,6 +99,8 @@ C_LoadModule(const char *mspec, CK_FUNCTION_LIST_PTR_PTR funcs)
 			mod = NULL; /* already freed */
 	}
 failed:
+	if (mod && mod->handle)
```

Comment:
is the `mod` null check needed? I think at every code path we get here, the `mod` is not null. OTOH it should not hurt checking this as the code could change in the future.

Reply:
Just the `else` branch above `failed` sets `mod = NULL;` without explicit return.